server.js

```diff
@@ -20,4 +20,57 @@ app.get("/", function (request, response) {
   response.sendFile(__dirname + "/index.html");
 });
 
+app.get('/messages', function (request, response) {
+  response.json(messages)
+})
+
+app.post('/messages', function (request, response) {
+  let newMessage = {};
+
+  if (request.body.from.trim() && request.body.text.trim()) {
```

Comment:
This works well when the "from" and "text" fields are empty. If you send POST request without a body, an error will be thrown. Ideally you should be catching this scenario too.

Reply:
```suggestion
  if (request.body && request.body.from.trim() && request.body.text.trim()){
```
is this what you mean?